tls/s2n_kem.c

```diff
@@ -101,6 +101,33 @@ const struct s2n_kem_group *ALL_SUPPORTED_KEM_GROUPS[S2N_SUPPORTED_KEM_GROUPS_CO
  * functions may return any non-zero value to indicate failure. */
 #define GUARD_PQ_AS_RESULT(x) RESULT_ENSURE((x) == 0, S2N_ERR_PQ_CRYPTO)
 
+/* Checks if the Hybrid KEM is using the old Hybrid KEM format (PQ Length, PQ Share, ECC Length, ECC Share) or the new
+ * format (PQ Share, ECC Share). The new format will always be 4 bytes shorter than the old format, so we check if the
+ * total length is 4 bytes longer than expected for the new format, and if so, fall back to the old format. */
+int s2n_is_hybrid_kem_length_prefixed(const s2n_mode mode, uint16_t actual_hybrid_share_size, const struct s2n_kem_group *kem_group, bool *is_length_prefixed)
+{
```

Comment:
Can you add some safety checks? We're dereferencing a lot of pointers in here.

Reply:
Fixed